### PR TITLE
Add set default browser modal (Uplift to 1.20.x)

### PR DESCRIPTION
**Pull request by deeppandya**
Uplift of #7745 
Resolves https://github.com/brave/brave-browser/issues/12390

Approved, please ensure that before merging: 
- [x] You have checked CI and the builds, lint, and tests all pass or are not related to your PR. 
- [x] You have tested your change on Nightly. 
- [x] The PR milestones match the branch they are landing to. 

After you merge: 
- [ ] The associated issue milestone is set to the smallest version that the changes is landed on.

### CODE REVIEW

**Review by kjozwiak — APPROVED** [review]
Uplift into `1.20.x` approved after deliberating with @brave/uplift-approvers. QA has verified the PR on Nightly as per https://github.com/brave/brave-core/pull/7745#issuecomment-771564148.

**Comment by bsclifton** [comment]
Because translations aren't available (see https://github.com/brave/brave-browser/issues/13958), we should revert this and land with 1.21

**Comment by bsclifton** [comment]
Moved to 1.21 👍

**Pull request merged**
